tailnet/test/integration/integration.go

```diff
@@ -171,6 +181,24 @@ func (o SimpleServerOptions) Router(t *testing.T, logger slog.Logger) *chi.Mux {
 		r.Get("/latency-check", func(w http.ResponseWriter, r *http.Request) {
 			w.WriteHeader(http.StatusOK)
 		})
+		r.Post("/restart", func(w http.ResponseWriter, r *http.Request) {
+			logger.Info(r.Context(), "killing DERP server")
+			derpServer.Close()
+			derpServer = derp.NewServer(key.NewNode(), tailnet.Logger(logger.Named("derp")))
```

Comment:
This would be a good candidate for an atomic.Pointer to allow for atomically swapping them out. I'd also grab a reference to the old one first, store the new one, THEN close the old one. Otherwise clients could race the swap and still try to connect to the old one